dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/domain/batching/DataProcessorFactory.kt

```diff
@@ -6,49 +6,43 @@
 
 package com.datadog.android.core.internal.domain.batching
 
+import com.datadog.android.core.internal.data.Writer
 import com.datadog.android.core.internal.data.file.FileOrchestrator
 import com.datadog.android.core.internal.data.file.ImmediateFileWriter
-import com.datadog.android.core.internal.domain.FilePersistenceConfig
 import com.datadog.android.core.internal.domain.Serializer
 import com.datadog.android.core.internal.domain.batching.processors.DataProcessor
 import com.datadog.android.core.internal.domain.batching.processors.DefaultDataProcessor
 import com.datadog.android.core.internal.domain.batching.processors.NoOpDataProcessor
 import com.datadog.android.privacy.TrackingConsent
-import java.io.File
 import java.util.concurrent.ExecutorService
 
 internal class DataProcessorFactory<T : Any>(
-    private val intermediaryFolderPath: String,
-    private val targetFolderPath: String,
-    private val filePersistenceConfig: FilePersistenceConfig,
+    private val intermediaryFileOrchestrator: FileOrchestrator,
+    private val targetFileOrchestrator: FileOrchestrator,
     private val serializer: Serializer<T>,
-    private val executorService: ExecutorService
+    private val separator: CharSequence,
+    private val executorService: ExecutorService,
 ) {
 
     fun resolveProcessor(consent: TrackingConsent): DataProcessor<T> {
         return when (consent) {
             TrackingConsent.PENDING -> {
+                intermediaryFileOrchestrator.reset()
                 DefaultDataProcessor(
                     executorService,
-                    ImmediateFileWriter(buildFileOrchestrator(intermediaryFolderPath), serializer)
+                    ImmediateFileWriter(intermediaryFileOrchestrator, serializer, separator)
                 )
             }
             TrackingConsent.GRANTED -> {
+                targetFileOrchestrator.reset()
```

Comment:
We should not reset it for the authorised folder as this is shared with the Uploader and it needs to stay consistent (keeping count of the current files, prevFile, etc)